beprof/curve.py

```diff
@@ -143,6 +144,68 @@ def rebinned(self, step=0.1, fixp=0):
+
+        # first: if the user doesn't want to create a new object in memory, 2 possible options:
+        if not newobj:
+            if a2 >= a1 and b2 <= b1:
+                # if the orginal domain includes c2 domain, subtraction can be done as follows:
+                # as the output domain is gonna be self.x, the subtracted values should be
+                # interpolated c2 values for every point that is in self.x and in section [a2, b2]
+                # and defval for the others. Values to subtract can be calculated using np.interp:
+
+                y = np.interp(self.x, curve2.x, curve2.y, left=defval, right=defval)
+                # print('y: ', y, type(y))
+                # print('Y: ', self.y, type(self.y))
+                self.y =  self.y - y
+                return
+            else:
+                # if the orginal domain doesn't include c2 domain there is nothing that can be done
+                print('Error, can not subtract without creating new object')
+                return
+
+        # since the program is here, that means newobj is True
+        # what means the method is not going to modify self but create new curve object instead
+        #
+        # the first thing: what should be the domain of a new object?
+        # User can set domain parameter to:
+        # 'keep' - new object domain is self.x
+        # 'extend' - new object domain is sum of sets self.x, c2.x,
+        #  and values of c2.y in self.x-c2.x and self.x in c2.x - self.x are defval
+
+        # anyway, there are 3 main possibilities:
+        # 1) domains are disjoint when a2>b1 or a1>b2
+        if a2 > b1 or a1 > b2:
+            #if user decides to keep the orignal domain, the new object should be as follows:
+            if domain == 'keep':
+                obj = copy.deepcopy(self)
+                obj.y = obj.y - defval
+                return obj
+            # could be else here but there might be more possibilities in the future
+            # if user wants to extend the domain, new domain is a union.
+            if domain == 'extend':
+                tmp1 = np.array(self.y) - defval        # new values for self.x are correspoding self.y-defval
+                tmp2 = defval - np.array(curve2.y)          # new values for c2.x are corresponding defval - c2.y
+                if a2 > b1:     # here we have ------a1||||||b1-----a2|||||b2------>
+                    newX = np.concatenate((self.x, curve2.x))        # new X is just an union
+                    newY = np.concatenate((tmp1, tmp2))    # into one list
+                else:           # mirrored, just a minor change
+                    newX = np.concatenate((curv2.x, self.x))        # new X is just an union
+                    newY = np.concatenate((tmp2, tmp1))             # into one list
+
+                print('newX: ', newX)
+                print('newY: ', newY)
+                obj = Curve(np.stack((newX, newY), axis=1), **self.__dict__['metadata'])
```

Comment:
Will this work if I try to subtract two profiles ?
```
p1 = profile.Profile(...)
p2 = profile.Profile(...)
r = p1.subtract(p2)
```

What will be the type of r ?